hbw/app/javascript/packs/hbw/components/form.js.jsx

```diff
@@ -216,9 +216,38 @@ modulejs.define('HBWForm', ['React', 'jQuery', 'HBWError', 'HBWFormDatetime',
         return result;
       }, {});
 
-      this.setState({ formValues: variables });
+      this.setState({ formValues: this.updateSelectVariables(variables) });
     };
 
+    updateSelectVariables = (variables) => {
+      const flattenFields = this.flatFields(this.props.form.fields);
+
+      const fieldNamesforUpdate = flattenFields.filter(
```

Comment:
i think you need separate function here

and please use `field.choices.length > 0`